src/hotspot/share/interpreter/interpreterRuntime.cpp

```diff
@@ -577,7 +577,7 @@ JRT_ENTRY(address, InterpreterRuntime::exception_handler_for_exception(JavaThrea
 
   // notify debugger of an exception catch
   // (this is good for exceptions caught in native methods as well)
-  if (JvmtiExport::can_post_on_exceptions()) {
+  if (JvmtiExport::can_post_on_exceptions() || JvmtiExport::can_post_frame_pop()) {
```

Comment:
This doesn't seem like it is related to ClearAllFramePops.

Reply:
Yes, it does not relate to the `ClearAllFramePops` directly. But it is needed for correctness of the `NotifyFramePop`. I do not feel it is worth to separate this change.

Reply:
So this is a long standing bug hidden by the fact that if `can_post_frame_pop()` is true, then so is `can_post_on_exceptions()`:

```
  JvmtiExport::set_can_post_on_exceptions(
    avail.can_generate_exception_events ||
    avail.can_generate_frame_pop_events ||
    avail.can_generate_method_exit_events);

  JvmtiExport::set_can_post_frame_pop(avail.can_generate_frame_pop_events);
```

Or you could argue that the existing code is correct because it already captures `can_post_frame_pop()`.

Reply:
Good check, thanks! Rolled back this change.
I remember that I've added this because some of the tests was failing. Most likely, I've made a wrong conclusion.